haystack/schema.py

```diff
@@ -521,11 +541,7 @@ def __init__(
         self.updated_at = updated_at
         self.query = query
 
-        if isinstance(answer, dict):
-            answer = Answer.from_dict(answer)
```

Comment:
This was moved to the `from_dict` method of `Label` since according to the schema `dict` is not a valid type for `Answer`. I believe this was done previously to get `Label.from_dict(label.to_dict())` to work so I don't anticipate this causing any breaking changes.